src/gossip/service.zig

```diff
@@ -145,36 +148,59 @@ pub const GossipService = struct {
     counter: *Atomic(usize),
     service_exit: Atomic(bool),
 
-    // communication between threads
+    /// communication between threads
     packet_incoming_channel: *Channel(Packet),
     packet_outgoing_channel: *Channel(Packet),
     verified_incoming_channel: *Channel(GossipMessageWithEndpoint),
 
-    // TODO(x19): most of these things need to be on the heap to be shared between threads
-    // but rn we do this by having GossipService be on the heap, which is not ideal
-    // ... we should figure out a better way to do this
+    /// table to store gossip values
     gossip_table_rw: RwMux(GossipTable),
-    // push message things
+    /// manages push message peers
     active_set_rw: RwMux(ActiveSet),
-    /// All gossip data pushed into this will have its wallclock overwritten during `drainPushQueueToGossipTable`.
+    /// all gossip data pushed into this will have its wallclock overwritten during `drainPushQueueToGossipTable`.
     push_msg_queue_mux: Mux(ArrayList(GossipData)),
-    // pull message things
+    /// hashes of failed gossip values from pull responses
     failed_pull_hashes_mux: Mux(HashTimeQueue),
 
-    /// This contact info is mutated by the buildMessages thread, so it must
-    /// only be read by that thread, or it needs a synchronization mechanism.
+    /// entrypoint peers to start the process of discovering the network
     entrypoints: ArrayList(Entrypoint),
-    ping_cache_rw: RwMux(*PingCache),
+    /// manages ping/pong heartbeats for the network
+    ping_cache_rw: RwMux(PingCache),
+    thread_pool: ThreadPool,
+    // TODO: fix when http server is working
+    // echo_server: EchoServer,
     logger: ScopedLogger(@typeName(GossipService)),
-    thread_pool: *ThreadPool,
-    echo_server: EchoServer,
-
     metrics: GossipMetrics,
 
     const Self = @This();
 
     const Entrypoint = struct { addr: SocketAddr, info: ?ContactInfo = null };
 
+    pub fn create(
```

Comment:
NIT: Given that init is not used anywhere else from what I can tell, perhaps we don't need the new method. Also it would almost be nice to indicate in the method name whether we are initialising a struct on the heap or the stack. There is something I don't like about Struct.init(...) returning !*Self or !Self arbitrarily throughout sig. Interested to here your thoughts

Reply:
the idea was 'create' would create objects on the heap, and 'init' on the stack, similar to the allocator interface which uses `allocator.create(T)` -- the same pattern is followed by the channel -- we could delete `init` but i think the separation makes things clear and its nice to have both

Reply:
Yeah great, I like that. I think there are some other init methods which return pointers so I was confused. This is great then!